sympy/ntheory/factor_.py

```diff
@@ -1157,6 +1157,11 @@ def factorint(n, limit=None, use_trial=True, use_rho=True, use_pm1=True,
     smoothness, smoothness_p, divisors
 
     """
+    if generator:
+        return ((k,v) for k,v in factorint(n, limit=limit, use_trial=use_trial,
+            use_rho=use_rho, use_pm1=use_pm1, use_ecm=use_ecm, verbose=verbose,
+            visual=visual, multiple=multiple, generator=False).items())
+
```

Comment:
I don't think that this would be the correct change because it is the most trivial way to implement the generator.
We would expect to implement a whole new efficient method that can generate the sequence one by one than generating as whole, or at least try to decompose the existing implementation to make it lazily evaluated.

Reply:
I'm not sure if I understand you, but this generator outputs one factor at a time by calling next() on it rather than outputting the entire dictionary of factors as a whole. May I please know how I can more efficiently implement this?

Reply:
There is no advantage in making an iterator from a concrete data structure like a dictionary because it means that the calculation is still not lazy. The purpose of the issue is to make it so that factorint quickly returns small factors before spending a long time working on large factors. Then in some situations it wouldn't be necessary to compute all the factors.

Reply:
Okay, would it be possible for us to have a chat about this over the sympy Glitter? https://gitter.im/sympy/sympy# I'm just concerned that this thread will become overfly filled with me trying to understand the problem.

Reply:
I don't use gitter. I recommend trying to fix a different issue. That one is not as easy as it might have seemed.